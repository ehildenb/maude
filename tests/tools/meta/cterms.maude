load ../../../contrib/tools/meta/cterms.maude

--- Module CTERM-SET
--- ================

select CTERM-SET .

--- Operator `_st_`
--- ---------------

reduce ('_+_['Y:Nat, 'X:Nat, 'Y:Nat] st 'Y:Nat ?= '_*_['0.Nat,'1.Nat] st 'Y:Nat ?= '_*_['0.Nat,'1.Nat])
    == ('_+_['Y:Nat, 'X:Nat, 'Y:Nat] st 'Y:Nat ?= '_*_['0.Nat,'1.Nat]) .

reduce ('_+_['Y:Nat, ('X:Nat st 'Z:Nat ?= '_*_['1.Nat, '0.Nat]), 'Z:Nat])
    == ('_+_['Y:Nat, 'X:Nat, 'Z:Nat] st 'Z:Nat ?= '_*_['1.Nat, '0.Nat]) .

reduce ('_+_['Y:Nat, 'X:Nat] st ('Y:Nat ?= '_+_['X:Nat, 'Z:NzNat])) ?= ('_*_['Y:Nat, 'X:Nat] st ('Y:Nat ?= '_*_['X:Nat, 'Z:NzNat]))
    == ('Y:Nat ?= '_*_['X:Nat,'Z:NzNat] /\ 'Y:Nat ?= '_+_['X:Nat,'Z:NzNat] /\ '_*_['Y:Nat,'X:Nat] ?= '_+_['Y:Nat,'X:Nat]) .

reduce ('_+_['Y:Nat,'X:Nat] st 'Y:Nat ?= '_+_['X:Nat,'Z:NzNat]) ?= '_*_['Y:Nat,'X:Nat]
    == ('_+_['Y:Nat,'X:Nat] ?= '_*_['Y:Nat,'X:Nat] /\ 'Y:Nat ?= '_+_['X:Nat,'Z:NzNat]) .
