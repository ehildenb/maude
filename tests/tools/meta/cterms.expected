==========================================
reduce in CTERM-SET : ('_+_['Y:Nat,'X:Nat,'Y:Nat] | 'Y:Nat ?= '_*_['0.Nat,
    '1.Nat] | 'Y:Nat ?= '_*_['0.Nat,'1.Nat]) == ('_+_['Y:Nat,'X:Nat,'Y:Nat] |
    'Y:Nat ?= '_*_['0.Nat,'1.Nat]) .
rewrites: 3
result Bool: true
==========================================
reduce in CTERM-SET : '_+_['Y:Nat,'X:Nat | 'Z:Nat ?= '_*_['1.Nat,'0.Nat],
    'Z:Nat] == ('_+_['Y:Nat,'X:Nat,'Z:Nat] | 'Z:Nat ?= '_*_['1.Nat,'0.Nat]) .
rewrites: 2
result Bool: true
Warning: sort declarations for operator resolveNames failed preregularity check
    on 6 out of 43 sort tuples. First such tuple is (Type).
Warning: sort declarations for operator resolveNames failed preregularity check
    on 1 out of 26 sort tuples. First such tuple is (NullDeclSet).
Warning: ctor declarations for associative operator __ are conflict on 138 out
    of 17576 sort triples. First such triple is (ModuleDeclSet, SortDeclSet,
    SortDeclSet).
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), '0.Nat) == '0.Nat .
rewrites: 10
result Bool: true
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), 'X:Nat) == 'X:Nat .
rewrites: 10
result Bool: true
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), '_+_['0.Nat,'1.Nat]) == '_+_['0.Nat,
    '1.Nat] .
rewrites: 10
result Bool: true
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), '_+_['0.Nat,'X:Nat]) == '_+_['0.Nat,
    'X:Nat] .
rewrites: 10
result Bool: true
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), '_*_['0.Nat,'1.Nat]) == (
    '#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat |
    '#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat ?= '_*_['0.Nat,'1.Nat]) .
rewrites: 35
result Bool: true
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), '_+_['_*_['0.Nat,'1.Nat],'X:Nat]) ==
    ('_+_['#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat,'X:Nat] |
    '#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat ?= '_*_['0.Nat,'1.Nat]) .
rewrites: 48
result Bool: true
==========================================
reduce in PURIFICATION : subtheoryPurify(asTemplate(upModule('FVP-NAT, true)),
    upModule('MULT-LOCALLY-FVP-NAT, true), '_+_['_*_['0.Nat,'1.Nat],'X:Nat,
    '_*_['0.Nat,'1.Nat]]) == ('_+_['#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat,
    'X:Nat,'#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat] |
    '#makeVariable`(_*_`[0.Nat`,1.Nat`]`):Nat ?= '_*_['0.Nat,'1.Nat]) .
rewrites: 82
result Bool: true
Warning: sort declarations for operator resolveNames failed preregularity check
    on 6 out of 43 sort tuples. First such tuple is (Type).
Warning: sort declarations for operator resolveNames failed preregularity check
    on 1 out of 26 sort tuples. First such tuple is (NullDeclSet).
Warning: ctor declarations for associative operator __ are conflict on 138 out
    of 17576 sort triples. First such triple is (ModuleDeclSet, SortDeclSet,
    SortDeclSet).
==========================================
reduce in TEST-PURIFY : 'Foo inS moduleIntersect(asTemplate(natOrderMod),
    asTemplate(natListMod)) == false .
rewrites: 26
result Bool: true
==========================================
reduce in TEST-PURIFY : 'NatList* inS moduleIntersect(asTemplate(natOrderMod),
    asTemplate(natListMod)) == false .
rewrites: 26
result Bool: true
==========================================
reduce in TEST-PURIFY : joint-variable(natListMod, natOrderMod, 'head[
    'L2:NeNatList]) .
rewrites: 44
result Variable: '#makeVariable`(head`[L2:NeNatList`]`):Nat
==========================================
reduce in TEST-PURIFY : joint-variable(natOrderMod, natListMod, 'head[
    'L2:NeNatList]) .
rewrites: 45
result Variable: '#makeVariable`(head`[L2:NeNatList`]`):Nat
==========================================
reduce in TEST-PURIFY : purify(natOrderMod, natListMod, '_<_['head[
    'L1:NeNatList],'head['L2:NeNatList]]) .
rewrites: 103
result CTerm: '_<_['#makeVariable`(head`[L1:NeNatList`]`):Nat,
    '#makeVariable`(head`[L2:NeNatList`]`):Nat] |
    '#makeVariable`(head`[L1:NeNatList`]`):Nat ?= 'head['L1:NeNatList] /\
    '#makeVariable`(head`[L2:NeNatList`]`):Nat ?= 'head['L2:NeNatList]
==========================================
reduce in TEST-PURIFY : purify(natListMod, natOrderMod, '_<_['head[
    'L1:NeNatList],'head['L2:NeNatList]]) .
rewrites: 164
result CTerm: '#makeVariable`(_<_`[#makeVariable`(head`[L1:NeNatList`]`):Nat`,#makeVariable`(head`[L2:NeNatList`]`):Nat`]`):Bool | '#makeVariable`(_<_`[#makeVariable`(head`[L1:NeNatList`]`):Nat`,#makeVariable`(head`[L2:NeNatList`]`):Nat`]`):Bool ?= '_<_['#makeVariable`(head`[L1:NeNatList`]`):Nat,
    '#makeVariable`(head`[L2:NeNatList`]`):Nat] /\
    '#makeVariable`(head`[L1:NeNatList`]`):Nat ?= 'head['L1:NeNatList] /\
    '#makeVariable`(head`[L2:NeNatList`]`):Nat ?= 'head['L2:NeNatList]
==========================================
reduce in TEST-PURIFY : purify(natListMod, natOrderMod, '_<_['_+_['head[
    'L':NeNatList],'head['L':NeNatList]],'_+_['head['L:NeNatList],'head[
    'L:NeNatList]]]) .
rewrites: 300
result CTerm: '#makeVariable`(_<_`[_+_`[#makeVariable`(head`[L':NeNatList`]`):Nat`,#makeVariable`(head`[L':NeNatList`]`):Nat`]`,_+_`[#makeVariable`(head`[L:NeNatList`]`):Nat`,#makeVariable`(head`[L:NeNatList`]`):Nat`]`]`):Bool | '#makeVariable`(_<_`[_+_`[#makeVariable`(head`[L':NeNatList`]`):Nat`,#makeVariable`(head`[L':NeNatList`]`):Nat`]`,_+_`[#makeVariable`(head`[L:NeNatList`]`):Nat`,#makeVariable`(head`[L:NeNatList`]`):Nat`]`]`):Bool ?= '_<_['_+_[
    '#makeVariable`(head`[L':NeNatList`]`):Nat,
    '#makeVariable`(head`[L':NeNatList`]`):Nat],'_+_[
    '#makeVariable`(head`[L:NeNatList`]`):Nat,
    '#makeVariable`(head`[L:NeNatList`]`):Nat]] /\
    '#makeVariable`(head`[L':NeNatList`]`):Nat ?= 'head['L':NeNatList] /\
    '#makeVariable`(head`[L:NeNatList`]`):Nat ?= 'head['L:NeNatList]
==========================================
reduce in TEST-PURIFY : purify(natOrderMod, natMultMod, deepLeft) .
rewrites: 114
result CTerm: '#makeVariable`(_*_`[1.Nat`,min`[0.Nat`,_*_`[1.Nat`,max`[0.Nat`,_*_`[1.Nat`,_+_`[1.Nat`,1.Nat`]`]`]`]`]`]`):Nat | '#makeVariable`(_*_`[1.Nat`,min`[0.Nat`,_*_`[1.Nat`,max`[0.Nat`,_*_`[1.Nat`,_+_`[1.Nat`,1.Nat`]`]`]`]`]`]`):Nat
    ?= '_*_['1.Nat,'min['0.Nat,'_*_['1.Nat,'max['0.Nat,'_*_['1.Nat,'_+_['1.Nat,
    '1.Nat]]]]]]
==========================================
reduce in TEST-PURIFY : purify(natMultMod, natOrderMod, deepLeft) .
rewrites: 7
result GroundTerm: '_*_['1.Nat,'min['0.Nat,'_*_['1.Nat,'max['0.Nat,'_*_['1.Nat,
    '_+_['1.Nat,'1.Nat]]]]]]
==========================================
reduce in TEST-PURIFY : purify(natOrderMod, natMultMod, deepRight) .
rewrites: 114
result CTerm: '#makeVariable`(_*_`[min`[_*_`[max`[_*_`[_+_`[1.Nat`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`):Nat | '#makeVariable`(_*_`[min`[_*_`[max`[_*_`[_+_`[1.Nat`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`):Nat
    ?= '_*_['min['_*_['max['_*_['_+_['1.Nat,'1.Nat],'1.Nat],'1.Nat],'1.Nat],
    '1.Nat],'1.Nat]
==========================================
reduce in TEST-PURIFY : purify(natOrderMod, natMultMod, deepMixed) .
rewrites: 193
result CTerm: '#makeVariable`(_+_`[_*_`[min`[_*_`[max`[_*_`[_+_`[1.Nat`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,_*_`[1.Nat`,min`[0.Nat`,_*_`[1.Nat`,max`[0.Nat`,_*_`[1.Nat`,_+_`[1.Nat`,1.Nat`]`]`]`]`]`]`]`):Nat | '#makeVariable`(_+_`[_*_`[min`[_*_`[max`[_*_`[_+_`[1.Nat`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,1.Nat`]`,_*_`[1.Nat`,min`[0.Nat`,_*_`[1.Nat`,max`[0.Nat`,_*_`[1.Nat`,_+_`[1.Nat`,1.Nat`]`]`]`]`]`]`]`):Nat ?= '_+_['_*_['min['_*_['max['_*_['_+_[
    '1.Nat,'1.Nat],'1.Nat],'1.Nat],'1.Nat],'1.Nat],'1.Nat],'_*_['1.Nat,'min[
    '0.Nat,'_*_['1.Nat,'max['0.Nat,'_*_['1.Nat,'_+_['1.Nat,'1.Nat]]]]]]]
==========================================
reduce in TEST-PURIFY : purify('MYLISTNAT, 'PRED-FVP-NAT, 'tt.Bool ?= '_<_[
    'head['L1:NeNatList],'head['L2:NeNatList]]) .
rewrites: 207
result PosEqConj: '#makeVariable`(head`[L1:NeNatList`]`):Nat ?= 'head[
    'L1:NeNatList] /\ '#makeVariable`(head`[L2:NeNatList`]`):Nat ?= 'head[
    'L2:NeNatList] /\ 'tt.Bool ?= '_<_[
    '#makeVariable`(head`[L1:NeNatList`]`):Nat,
    '#makeVariable`(head`[L2:NeNatList`]`):Nat]
==========================================
reduce in TEST-PURIFY : purify('PRED-FVP-NAT, 'MYLISTNAT, 'tt.Bool != '_<_[
    '_+_['head['L':NeNatList],'head['L':NeNatList]],'_+_['head['L:NeNatList],
    'head['L:NeNatList]]]) .
rewrites: 321
result EqConj: '#makeVariable`(head`[L':NeNatList`]`):Nat ?= 'head[
    'L':NeNatList] /\ '#makeVariable`(head`[L:NeNatList`]`):Nat ?= 'head[
    'L:NeNatList] /\ 'tt.Bool != '_<_['_+_[
    '#makeVariable`(head`[L':NeNatList`]`):Nat,
    '#makeVariable`(head`[L':NeNatList`]`):Nat],'_+_[
    '#makeVariable`(head`[L:NeNatList`]`):Nat,
    '#makeVariable`(head`[L:NeNatList`]`):Nat]]
==========================================
reduce in TEST-PURIFY : purify('MYLISTNAT, 'PRED-FVP-NAT, 'tt.Bool != '_<_[
    '_+_['head['L':NeNatList],'head['L':NeNatList]],'_+_['head['L:NeNatList],
    'head['L:NeNatList]]]) .
rewrites: 321
result EqConj: '#makeVariable`(head`[L':NeNatList`]`):Nat ?= 'head[
    'L':NeNatList] /\ '#makeVariable`(head`[L:NeNatList`]`):Nat ?= 'head[
    'L:NeNatList] /\ 'tt.Bool != '_<_['_+_[
    '#makeVariable`(head`[L':NeNatList`]`):Nat,
    '#makeVariable`(head`[L':NeNatList`]`):Nat],'_+_[
    '#makeVariable`(head`[L:NeNatList`]`):Nat,
    '#makeVariable`(head`[L:NeNatList`]`):Nat]]
Bye.
